meta-resin-common/recipes-bsp/u-boot/patches/env_resin.h

```diff
@@ -79,9 +83,17 @@
                        "run resin_import_env_file;" \
                "fi;\0" \
        \
+       "resin_check_kernel_extroot=" \
+               "if run resin_kernel_extdetect; then " \
+                 "env set resin_kernel_loader ext4load; " \
+               "else; "\
+                 "env delete resin_kernel_loader; " \
+               "fi;\0" \
+       \
        "resin_set_kernel_root=" \
                "run resin_set_dev_index;" \
                "run resin_inject_env_file;" \
+               "run resin_check_kernel_extroot;" \
                "run resin_find_root_part_uuid;" \
```

Comment:
@agherzan  I'm wondering if this check is needed.
The original thought was to have a backwards compatible design where meta-resin can keep moving forward and update env_resin.h. It'll set an env var resin_kernel_loader.

And integration layers will add checks if  resin_kernel_loader exists. load kernel from ext. otherwise it'll default to trying to load from the boot partition.

But I'm not sure if this backwards compatibility is needed/has any benefit

Reply:
Which check are you referring to?

Reply:
I think you are referring to resin_check_kernel_extroot. I commented on the resin-rpi PR too. In short I agree. Don't see the benefit either.